README_AWS.md

```diff
@@ -0,0 +1,143 @@
+    - [1.2 Interpreting Output](#12-interpreting-output)
+    - [1.3 Workflow](#13-workflow)
+  - [2. BYOVPC Configurations Verification](#2-byovpc-configurations-verification)
+
+## Setup ##
+### AWS Environment ###
+Set up your environment to use the correct credentials for the AWS account for the target cluster. 
+- Obtain a valid set of AWS secret and key for the target account.
+- Export these AWS credentials:
+   ```shell
+   export AWS_ACCESS_KEY_ID=<YOUR_AWS_ACCESS_KEY_ID)>
+   export AWS_SECRET_ACCESS_KEY=<YOUR_AWS_SECRET_ACCESS_KEY>
+   ```
+  For STS credentials, also:
+    ```shell 
+    export AWS_SESSION_TOKEN=<YOUR_SESSION_TOKEN_STRING> 
+    ```
+  Export any other AWS environment vars:
+    ```shell
+    export AWS_REGION=<VPC_AWS_REGION>
+    ````
+  
+### VPC ###
+- Any VPC for a non-STS CCS cluster can be tested using this tool.
+- Apart from the AWS credentials, you will need to know the following information about the VPC to be verified.
+    - Subnet IDs
+    - AWS region
+  
+### IAM permissions ###
+Ensure that the AWS credentials being used have the following permissions.
+```json
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Effect": "Allow",
+      "Action": [
+        "ec2:RunInstances",
+        "ec2:DescribeInstanceStatus",
+        "ec2:DescribeInstanceTypes",
+        "ec2:GetConsoleOutput",
+        "ec2:TerminateInstances"
+      ],
+      "Resource": "*"
+    }
+  ]
+}
+```
+ 
+## Available Tools ##
+
+### 1. Egress Verification ###
+#### 1.1 Usage ####
+The processes below describe different ways of using egress verifier on a single subnet. 
+In order to verify entire VPC, 
+repeat the verification process for each subnet ID.
+
+##### 1.1.1 CLI Executable #####
+   1. Ensure correct [environment setup](#setup).
+
+   2. Clone the source:
+      ```shell
+      git clone git@github.com:openshift/osd-network-verifier.git
+      ``` 
+   3. Build the cli:
+      ```shell
+      make build
+      ```
+      This generates `osd-network-verifier` executable in project root directory. 
+
+   4. Obtain params:
+      1. subnet_id: Obtain the subnet id to be verified. 
+      2. image_id: Select an optional image id parameter (ami-xxxxxxxxxxxx) to run on ec2 instance. 
+      
+         You may use the following public image-id:
+         ```bash
+          resolve:ssm:/aws/service/ami-amazon-linux-latest/amzn2-ami-hvm-x86_64-gp2`
+         ```
+          If the image id is not provided, it is defaulted to an image id from [AWS account olm-artifacts-template.yaml](https://github.com/openshift/aws-account-operator/blob/17be7a41036e252d59ab19cc2ad1dcaf265758a2/hack/olm-registry/olm-artifacts-template.yaml#L75),
+   for the same region where your subnet is.
+
+   5. Execute:
+    
+        ```shell        
+        ./osd-network-verifier egress \
+            --subnet-id <subnet_id> \ 
+            --image-id <image_id>
+        ```
```

Comment:
(optional)I think we can also include an example for egress with just _--region_ and _--subnet-id_ flag which is also an option.
https://github.com/openshift/ops-sop/blob/master/v4/utils/cluster_provisioning_failure#L161

Reply:
Thanks! I changed this part to show just the required flags, adding a description of optional once. I'll add a new "examples" section to note how to use with different available flags (like --region) in different scenarios.